src/Fossa/API/Types.hs

```diff
@@ -353,6 +357,26 @@ instance FromJSON ScanResponse where
       <$> obj .: "id"
       <*> obj .:? "status"
 
+data RevisionDependencyCacheStatus
+  = Ready
+  | Waiting
+  | UnknownDependencyCacheStatus Text
+  deriving (Eq, Ord, Show)
+
+newtype RevisionDependencyCache = RevisionDependencyCache {status :: RevisionDependencyCacheStatus}
+  deriving (Eq, Ord, Show)
+
+instance FromJSON RevisionDependencyCache where
+  parseJSON = withObject "RevisionDependencyCache" $ \obj ->
+    RevisionDependencyCache
+      <$> obj .: "status"
+
+instance FromJSON RevisionDependencyCacheStatus where
+  parseJSON = withText "RevisionDependencyCacheStatus" $ \case
```

Comment:
[nit]: We should compare in a case-insensitive way, to be defensive against accidental server-side changes.
```suggestion
  parseJSON = withText "RevisionDependencyCacheStatus" $ \txt -> case Text.toUpper txt of
```